src/library_wasmfs.js

```diff
@@ -143,6 +143,25 @@ FS.createPreloadedFile = FS_createPreloadedFile;
       });
     },
     // TODO: read
+    read: (fd, buffer, offset, length, position) => {
+      var seeking = typeof position != 'undefined';
+
+      var dataBuffer = _malloc(length);
+
+      var bytesRead;
+      if (seeking) {
+        bytesRead = FS.handleError(__wasmfs_pread(fd, dataBuffer, length, position));
+      } else {
+        bytesRead = FS.handleError(__wasmfs_read(fd, dataBuffer, length));
+      }
```

Comment:
```suggestion
      if (seeking) {
        bytesRead = __wasmfs_pread(fd, dataBuffer, length, position);
      } else {
        bytesRead = __wasmfs_read(fd, dataBuffer, length);
      }
      bytesRead = FS.handleError(bytesRead);
```
The handleError logic can be factored out into a shared place, which makes this easier to read, I think - then the if just has the actually different things for read/pread.